lib/ui/painting/image_encoding_unittests.cc

```diff
@@ -78,5 +99,70 @@ TEST_F(ShellTest, EncodeImageGivesExternalTypedData) {
   DestroyShell(std::move(shell), std::move(task_runners));
 }
 
+TEST_F(ShellTest, EncodeImageAccessesSyncSwitch) {
+  Settings settings = CreateSettingsForFixture();
+  TaskRunners task_runners("test",                  // label
+                           GetCurrentTaskRunner(),  // platform
+                           CreateNewThread(),       // raster
+                           CreateNewThread(),       // ui
+                           CreateNewThread()        // io
+  );
+
+  auto nativeEncodeImage = [&](Dart_NativeArguments args) {
+    auto image_handle = Dart_GetNativeArgument(args, 0);
+    image_handle =
+        Dart_GetField(image_handle, Dart_NewStringFromCString("_image"));
+    ASSERT_FALSE(Dart_IsError(image_handle)) << Dart_GetError(image_handle);
+    ASSERT_FALSE(Dart_IsNull(image_handle));
+    auto format_handle = Dart_GetNativeArgument(args, 1);
+
+    intptr_t peer = 0;
+    Dart_Handle result = Dart_GetNativeInstanceField(
+        image_handle, tonic::DartWrappable::kPeerIndex, &peer);
+    ASSERT_FALSE(Dart_IsError(result));
+    CanvasImage* canvas_image = reinterpret_cast<CanvasImage*>(peer);
+
+    int64_t format = -1;
+    result = Dart_IntegerToInt64(format_handle, &format);
+    ASSERT_FALSE(Dart_IsError(result));
+
+    auto io_manager = UIDartState::Current()->GetIOManager();
+    fml::AutoResetWaitableEvent latch;
+
+    task_runners.GetIOTaskRunner()->PostTask([&]() {
+      const std::shared_ptr<const MockSyncSwitch>& is_gpu_disabled_sync_switch =
+          std::make_shared<MockSyncSwitch>();
```

Comment:
This variable shouldn't be a reference.  You can just use auto here `auto is_gpu_disabled_sync_switch = std::make_shared<MockSyncSwitch>();` since the type is in the right hand expression.

Reply:
Done